src/PSR7/SpecFinder.php

```diff
@@ -74,37 +108,49 @@ public function findPathSpec(OperationAddress $addr) : PathItem
         return $pathSpec;
     }
 
-    /**
-     * @return Parameter[]
-     *
-     * @throws NoPath
-     */
-    public function findPathSpecs(OperationAddress $addr) : array
+    private function findPathSimple(OperationAddress $addr) : ?PathItem
```

Comment:
I'd rather suggest to name it something like `findExactPath` and `findMatchingTemplatePath` or something like this. Current naming gives no real understanding what each method does under the hood.

Reply:
Also we can try incapsulate this method into PathFinder, keeping thing simpler.